modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -520,7 +520,7 @@ private <R> CompletableFuture<R> trackingInvoke(
             // Track only write requests from explicit transactions.
             if (!txManager.addInflight(tx.id())) {
                 return failedFuture(
-                        new TransactionException(TX_UNEXPECTED_STATE_ERR, format(
+                        new TransactionException(TX_FAILED_READ_WRITE_OPERATION_ERR, format(
```

Comment:
Should we name it TX_ENLISTMENT_ERR or similar in order not to update it when we will expand the logic over RO transactions?

Reply:
I guess not, because this error is possible on PartitionReplicaListener side (after enlistment) and we dont enlist anything into RO transactions as well